packages/http-server/src/server.ts

```diff
@@ -75,14 +91,22 @@ const replyHandler = <LoaderInput>(
           reply.headers(output.headers);
         }
 
-        reply.serializer((payload: unknown) => payload);
-        reply.send(output.body);
+        reply.serializer((payload: unknown) => payload).send(output.body);
       } else {
         reply.code(500).send('Unable to find any decent response for the current request.');
       }
     } catch (e) {
-      reply.code(500).send(e);
+      const status = 'status' in e ? e.status : 500;
+      reply
+        .type('application/problem+json')
+        .serializer(problemJsonSerializer)
+        .code(status)
+        .send({
+          type: e.name || 'https://stoplight.io/prism/errors#UNKNOWN',
+          title: e.message,
+          status,
+          detail: e.detail || '',
```

Comment:
Isn't `detail` set to empty by default in the JsonProblemError constructor?

Reply:
Good catch! cb18613]